src/firefly/java/edu/caltech/ipac/table/io/VoTableWriter.java

```diff
@@ -0,0 +1,443 @@
+                                                        VOSerializer.formatAttribute(TableMeta.NAME, oneAtt.getKey()) +
+                                                        VOSerializer.formatAttribute("value", oneAtt.getValue()) + "/>" )
+                                                .collect(Collectors.joining("\n"));
+        }
+
+        public String xmlDESCRIPTION() {
+            String descVal = getMetaValueOf(TableMeta.DESC);
+
+            return isEmpty(descVal) ? "" : "<"+tagDesc+">" + descVal + "</"+tagDesc+">";
+        }
+
+        private String xmlLINK(LinkInfo link) {
+            List<String> atts = new ArrayList<>();
+
+            atts.add(elementAtt(TableMeta.ID, link.getID()));
+            atts.add(elementAtt("content-role", link.getRole()));
+            atts.add(elementAtt("content-type", link.getType()));
+            atts.add(elementAtt("title" , link.getTitle()));
+            atts.add(elementAtt("href", link.getHref()));
+
+            String attsInElement =  atts.stream()
+                                    .filter( v -> !isEmpty(v))
+                                    .collect(Collectors.joining());
+
+            return "<LINK"+attsInElement + "/>";
+        }
+
+        private String xmlVALUES(String nullStr, String minStr, String maxStr, String options) {
+            String[] opts = isEmpty(options) ? new String[0] : options.split(",");
+
+            if (isEmpty(nullStr) && isEmpty(minStr) && isEmpty(maxStr) && (opts.length == 0)) {
+                return "";
+            }
+
+            //  null="null", null='""' or null="xxxx"
+            String valuesAtts = elementAtt("null", nullStr);
+            String valuesElement = "";
+            if (!isEmpty(minStr)) {
+                valuesElement += "<MIN" + elementAtt("value", minStr) + "/>\n";
+            }
+            if (!isEmpty(maxStr)) {
+                valuesElement += "<MAX" + elementAtt("value", maxStr) + "/>\n";
+            }
+            if (opts.length > 0) {
+                for (String op : opts) {
+                    valuesElement += "<OPTION" + elementAtt("value", op) + "/>\n";
+                }
+            }
+
+            return completeTag(valuesAtts, valuesElement, "VALUES");
+        }
+
+        private String xmlColumnAtt(DataType dt) {
+            String prec = dt.getPrecision();
+            String width = dt.getWidth() == 0 ? "" : ""+dt.getWidth();
+            String atts = elementAtt(TableMeta.ID, dt.getID()) +
+                          elementAtt(TableMeta.NAME, dt.getKeyName()) +
+                          elementAtt(TableMeta.UCD, dt.getUCD()) +
+                          elementAtt("datatype", dataTypeMap.get(dt.getDataType())) +
+                          getArraySize(dt.getDataType()) +
+                          elementAtt("width", width) +
+                          elementAtt("precision",
+                                     (!isEmpty(prec) && prec.startsWith("F")) ? prec.substring(1) : prec) +
```

Comment:
No need to remove the `F` from precision.  Having the `F` is just as good.
You may need to remove the `G`, because it's not a votable standard.

Reply:
ok